tests/unit/test_xapidb_filter.py

```diff
@@ -55,9 +55,34 @@
 """
 
 
+def assert_xml_element_trees_equiv(a, b):
+    """Assert that the contents of two XML ElementTrees are equivalent (recursive)"""
+    # Check the XML tag
+    assert a.tag == b.tag
+    # Check the XML text
+    assert (a.text or "").strip() == (b.text or "").strip()
+    # Check the XML tail
+    assert (a.tail or "").strip() == (b.tail or "").strip()
+    # Check the XML attributes
+    assert a.attrib.items() == b.attrib.items()
+    # Compare the number of child nodes
+    assert len(list(a)) == len(list(b))
+    # Recursively repeat for all child nodes (expect same order of child nodes):
+    for achild, bchild in zip(list(a), list(b)):
+        assert assert_xml_element_trees_equiv(achild, bchild)
+    return True
```

Comment:
Why are we returning True (or at all)?
It's not possible for this function to return anything else, I'd expect usages to be isolated lines
`assert_xml_element_trees_equiv(...)`
The assert keyword on line 83 will always pass (because the method itself will raise an exception when an assertion fails)

Reply:
@GeraldEV applied, thanks!